Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -11,8 +11,57 @@
 import Foundation
 import SymbolKit
 
+typealias OverloadDeclaration = (
+    declaration: [SymbolGraph.Symbol.DeclarationFragments.Fragment],
+    reference: ResolvedTopicReference
+)
+
 /// Translates a symbol's declaration into a render node's Declarations section.
 struct DeclarationsSectionTranslator: RenderSectionTranslator {
+    /// A mapping from a symbol reference to the "common fragments" in its overload group.
+    static let commonFragmentsMap: Synchronized<[ResolvedTopicReference: [SymbolGraph.Symbol.DeclarationFragments.Fragment]]> = .init([:])
+
+    /// Set the common fragments for the given symbol references.
+    func setCommonFragments(
+        references: [ResolvedTopicReference],
+        fragments: [SymbolGraph.Symbol.DeclarationFragments.Fragment])
+    {
+        Self.commonFragmentsMap.sync({ map in
+            for reference in references {
```

Comment:
Why bother saving the fragments for all of the overload declarations? Reading this, I'm surprised that `setCommonFragments` takes an array of references, but `commonFragments` only queries one.

Reply:
The entire trick being relied upon here is that the list of common fragments is going to be the same for a group of overloaded declarations. So the first one in a group that gets rendered can save the calculation for the whole group. So when we check for a cached result, we only need to check for one of the declarations.

Reply:
So the `mainDeclaration` will change depending on when we call `commonFragments` below? Thanks for the explanation.